### ISSUE TITLE
CA1416 (PlatformCompatibilityAnalyzer) false positive

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#6189]
### Analyzer

**Diagnostic ID**: [CA1416](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1416)

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.0-preview1.22403.2

### Describe the bug

`CheckSystemVersion` calls are not taken into account.

### Steps To Reproduce

Sorry, not a complete minimal repro, but I hope the idea is clear. [`CheckSystemVersion`](https://learn.microsoft.com/en-us/dotnet/api/uikit.uidevice.checksystemversion?view=xamarin-ios-sdk-12#uikit-uidevice-checksystemversion(system-int32-system-int32)) calls are not taken into account.

```csharp
		private static (UIWindow[] Windows, CGRect StatusBarFrame) GetWindowsAndStatusBarFrame()
		{
			if (UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
			{
				IEnumerable<UIScene> scenes = UIApplication.SharedApplication.ConnectedScenes;
				var currentScene = scenes.FirstOrDefault(n => n.ActivationState == UISceneActivationState.ForegroundActive);

				if (currentScene is not UIWindowScene uiWindowScene)
					throw new InvalidOperationException("Unable to find current window scene.");

				if (uiWindowScene.StatusBarManager is not { } statusBarManager)
					throw new InvalidOperationException("Unable to find a status bar manager.");

				return (uiWindowScene.Windows, statusBarManager.StatusBarFrame);
			}
			else
			{
				return (UIApplication.SharedApplication.Windows, UIApplication.SharedApplication.StatusBarFrame);
			}
		}
```

### Expected behavior

No warnings.

### Actual behavior

```
Error CA1416: This call site is reachable on: 'MacCatalyst' 15.4 and later. 'UIApplication.StatusBarFrame.get' is unsupported on: 'maccatalyst' 13.0.0 and later.
Error CA1416: This call site is reachable on: 'MacCatalyst' 15.4 and later. 'UIApplication.Windows.get' is unsupported on: 'maccatalyst' 15.0.0 and later.
```

**Comment by elachlan** [comment]
I am also getting a false positive:
```csharp
[SupportedOSPlatform("windows8.1")]
        public static void test()
        {
        }

        public static void test2()
        {
            if (OperatingSystem.IsWindowsVersionAtLeast(6, 3))
            {
                test();
            }
        }
```

**Comment by buyaa-n** [comment]
> CheckSystemVersion calls are not taken into account.

@Youssef1313 by default the analyzer knows only knows group of platform check methods in `OperatingSystem` type and `.IsOSPlatform(OSPlatform)` method in `RuntimeInformation`

In order to be accounted as platform check method `UIDevice.CurrentDevice.CheckSystemVersion(13, 0)` needs to be annotated with corresponding guard method `SupportedOSPlatformGuard("platform")` or `UnsupportedOSPlatformGuard("platform")`

**Comment by buyaa-n** [comment]
`I am also getting a false positive:`

@elachlan  the `test` method is only supported on windows from version 8.1, therefore it could not called from within windows 6.3 context, because the call site is reachable on unsupported versions from 6.3 to 8.1

**Comment by elachlan** [comment]
> `I am also getting a false positive:`
> 
> @elachlan  the `test` method is only supported on windows from version 8.1, therefore it could not called from within windows 6.3 context, because the call site is reachable on unsupported versions from 6.3 to 8.1

6.2 is windows 8 and 6.3 is windows 8.1

**Comment by buyaa-n** [comment]
> 6.2 is windows 8 and 6.3 is windows 8.1

Well then the attribute should use the same version, `[SupportedOSPlatform("windows6.3")]` instead of `[SupportedOSPlatform("windows8.1")]`
